pyqentangle/schmidt.py

```diff
@@ -59,10 +59,13 @@ def schmidt_decomposition(bipartitepurestate_tensor):
     state_dims = bipartitepurestate_tensor.shape
     mindim = np.min(state_dims)
 
-    vecs1, diags, vecs2 = np.linalg.svd(bipartitepurestate_tensor)
-
-    decomposition = [(diags[k]*diags[k], vecs1[:, k], vecs2[:, k])
+    vecs1, diags, vecs2_h = np.linalg.svd(bipartitepurestate_tensor)
+    vecs2 = vecs2_h.transpose()
+    #decomposition = [(diags[k]*diags[k], vecs1[:, k], transposevecs2[:, k])
+    #                 for k in range(mindim)]
+    decomposition = [(diags[k], vecs1[:, k], vecs2[:, k])
```

Comment:
Why do you want to make such a change to the Schmidt coefficients?
I noticed your other changes are consistent with this change.

This is logically okay, but it is breaking all the unit tests.

Reply:
Well, the problem is that we are using SVD method in python and what it returns as it is "**s^2**" that gives the eigenvalues and not "**s**" itself. We should be careful when we are normalizing these coefficients because it is actually the sum over all the eigenvalues that should be equal to one, i.e. sum(s^2).

Reply:
So let me confirm: ![\Large |\psi \rangle = \sum_i \lambda_i |u_i\rangle |v_i \rangle ](<a href="https://www.codecogs.com/eqnedit.php?latex=|\psi&space;\rangle&space;=&space;\sum_i&space;\lambda_i&space;|u_i\rangle&space;|v_i&space;\rangle" target="_blank"><img src="https://latex.codecogs.com/svg.latex?|\psi&space;\rangle&space;=&space;\sum_i&space;\lambda_i&space;|u_i\rangle&space;|v_i&space;\rangle" title="|\psi \rangle = \sum_i \lambda_i |u_i\rangle |v_i \rangle" /></a>)

Hence, the s that you have is lambda. For me, sqrt(s) is lambda. Am I right?

Reply:
Oh, Sorry I just saw you changed back.